File: app/src/main/java/sample/jbanse/demosql/data/dao/NewsDetail.java
package sample.jbanse.demosql.data.dao;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.google.auto.value.AutoValue;

/**
 * Created by julien on 20/09/2017.
 */
@AutoValue
public abstract class NewsDetail implements NewsModel.SelectNewsDetailModel<News, Author> {

    public static final NewsModel.SelectNewsDetailMapper<News, Author, NewsDetail> MAPPER = News.FACTORY.selectNewsDetailMapper(new NewsModel.SelectNewsDetailCreator<News, Author, NewsDetail>() {
        @Override
        public NewsDetail create(@NonNull News news, @Nullable Author author) {
            return new AutoValue_NewsDetail(news, author);
        }
    }, Author.FACTORY);
}

File: app/src/main/java/sample/jbanse/demosql/data/dao/NewsDetail.kt
package sample.jbanse.demosql.data.dao

import com.google.auto.value.AutoValue

/**
 * Created by julien on 20/09/2017.
 */
@AutoValue
abstract class NewsDetail : NewsModel.SelectNewsDetailModel<News, Author> {
    companion object {
        @JvmField
        val MAPPER: NewsModel.SelectNewsDetailMapper<News, Author, NewsDetail> = News.FACTORY.selectNewsDetailMapper({ news, author -> AutoValue_NewsDetail(news, author) }, Author.FACTORY)
    }
}
